Sources/Fuzzilli/Core/ProgramBuilder.swift

```diff
@@ -377,31 +377,25 @@ public class ProgramBuilder {
 
     // This expands and collects types for arguments in function signatures.
     private func prepareArgumentTypes(forSignature signature: FunctionSignature) -> [Type] {
-        var parameterTypes = signature.inputTypes
+        var callerTypes = signature.callerTypes
         var argumentTypes = [Type]()
 
-        // "Expand" varargs parameters first
         if signature.hasVarargsParameter() {
-            let varargsParam = parameterTypes.removeLast()
-            assert(varargsParam.isList)
+            let restType = callerTypes.removeLast()
             for _ in 0..<Int.random(in: 0...5) {
-                parameterTypes.append(varargsParam.removingFlagTypes())
+                callerTypes.append(restType)
             }
         }
 
-        for var param in parameterTypes {
-            if param.isOptional {
+        for type in callerTypes {
+            if type == .undefined {
```

Comment:
This doesn't quite work, does it? The type would be a union with `.undefined`. But I think it'd anyway be better to call `isOptional` here instead.